utils/resultstable.go

```diff
@@ -1048,3 +1060,50 @@ func extractDependencyNameFromComponent(key string, techIdentifier string) (depe
 	dependencyName = split[0]
 	return
 }
+
+func getApplicabilityStatusFromRule(rule *sarif.ReportingDescriptor) ApplicabilityStatus {
+	if rule.Properties["applicability"] != nil {
+		status, ok := rule.Properties["applicability"].(string)
+		if !ok {
+			log.Debug(fmt.Sprintf("failed to get applicability status from rule properties for rule_id %s", rule.ID))
+		}
+		switch status {
+		case "not_covered":
+			return NotCovered
+		case "undetermined":
+			return ApplicabilityUndetermined
+		case "not_applicable":
+			return NotApplicable
+		case "applicable":
+			return Applicable
+		}
+	}
+	return ""
+}
+
+// If at least one status is applicable - final value is applicable
+// Else if at least one status is undetermined - final value is undetermined
+// Else if at least one status is not covered -> final value is not covered
+// Else (case when all statuses are not applicable) -> final value is not applicable
+func setApplicabilityStatusFromRule(applicabilityStatuses []ApplicabilityStatus) string {
```

Comment:
```suggestion
func getFinalApplicabilityStatus(applicabilityStatuses []ApplicabilityStatus) string {
```
A more fitting name for this method